R/deleteGates.R

```diff
@@ -0,0 +1,39 @@
+#' Delete gate
+#'
+#' Deletes a gate or a tailored gate family
+#' Compound gates (quadrant and split) are made up of "sectors." 
+#' Quadrant gates have four sectors (upper-right, upper-left, lower-left, lower-right) and split gates have two sectors (left and right).
+#' In addition to the top-level GID (like simple gates), these gates have model.gids and names lists that specify the GID and name for each sector, in the order shown above. 
+#' Populations using compound gates must reference these sector GIDs; referencing the top-level GID of a compound gate is meaningless.
+#' 
+#'
+#' @param experimentId ID of experiment.
+#' @param gid ID of gate family.
+#' @param gateId ID of gate.
+#' @param exclude Gate ID to exclude from deletion.
+#' @export
+#' @examples
+#' \dontrun{
+#' deleteGate(experimentId, gateId)
+#' }
+#' 
+#' TODO:(ge) This needs to account for compound gates (Quadrant, Split)
+
+deleteGates = function(experimentId, gid, gateId = NULL, exclude = NULL) {
+  checkDefined(experimentId)
+  experimentId = lookupByName("experiments", experimentId)
+  #checkDefined(gateId)
+  #gateId = lookupByName(paste("experiments", experimentId, "gates", sep = "/"), gateId, "filename")
```

Comment:
Maybe add a note to the docs that `byName` does not work for this function yet.

You *could* add it inside the `if (!is.null(gateId))` block, but it will error out when gates are tailored (which is correct behavior -- tailored gates all have the same name), and when trying to delete compound gates (`names` vs `name`). That'd address simple usages, but no pressure to do it.

A better, long-term solution (not this PR) would be a helper function like `findGateForFile(fcsFileId=NULL, gid=NULL, name=NULL)` helper function that picks a gate by name and FCS file, and supports compound gate names.